File: algorithms/0242.valid_anagram/valid_anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }
        let mut alpha_table = [0; 256];
        for (&b1, &b2) in s.as_bytes().iter().zip(t.as_bytes().iter()) {
            alpha_table[b1 as usize] += 1;
            alpha_table[b2 as usize] -= 1;
        }
        alpha_table.iter().all(|&x| x == 0)
    }

}

File: algorithms/0242.valid_anagram/valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        int s_map[256] = {0};
        int t_map[256] = {0};
        for (const char &c: s) {
            s_map[c] += 1;
        }
        for (const char &c: t) {
            t_map[c] += 1;
        }
        for (int i = 0; i < 256; ++i) {
            if (s_map[i] != t_map[i]) {
                return false;
            }
        }
        return true; 
    }
};